services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/FlowRulesSpec.groovy

```diff
@@ -100,19 +100,36 @@ class FlowRulesSpec extends HealthCheckSpecification {
 
         when: "Delete rules from the switch"
         def expectedRules = data.getExpectedRules(srcSwitch, srcSwDefaultRules)
-        def deletedRules = northbound.deleteSwitchRules(srcSwitch.dpId, data.deleteRulesAction)
-
-        then: "The corresponding rules are really deleted"
-        if (northbound.getSwitchProperties(srcSwitch.dpId).multiTable ) {
-            def ingressRule = (northbound.getSwitchRules(srcSwitch.dpId).flowEntries - expectedRules).find {
-                Cookie.isDefaultRule(it.cookie)
-            }
-            if (ingressRule) {
-                expectedRules = (expectedRules + ingressRule)
+        def isMultiTableEnabled = northbound.getSwitchProperties(srcSwitch.dpId).multiTable
+        if (isMultiTableEnabled) {
+            def extraIngressFlowRule
+            Wrappers.wait(RULES_INSTALLATION_TIME) {
+                extraIngressFlowRule = (northbound.getSwitchRules(srcSwitch.dpId).flowEntries - expectedRules).findAll {
+                    Cookie.isIngressRulePassThrough(it.cookie)
+                }
+                assert extraIngressFlowRule.size() == multiTableFlowRulesCount
             }
 
+            if (data.deleteRulesAction in [DeleteRulesAction.DROP_ALL_ADD_DEFAULTS,
+                                           DeleteRulesAction.IGNORE_DEFAULTS,
+                                           DeleteRulesAction.REMOVE_ADD_DEFAULTS,
+                                           DeleteRulesAction.OVERWRITE_DEFAULTS]) {
+                expectedRules = (expectedRules + extraIngressFlowRule)
+            }
         }
-        deletedRules.size() == data.rulesDeleted
+        def deletedRules = northbound.deleteSwitchRules(srcSwitch.dpId, data.deleteRulesAction)
+
+        then: "The corresponding rules are really deleted"
+        def amountOfDeletedRules
+        if (isMultiTableEnabled && data.deleteRulesAction in [DeleteRulesAction.DROP_ALL,
+                                                              DeleteRulesAction.DROP_ALL_ADD_DEFAULTS,
+                                                              DeleteRulesAction.REMOVE_DEFAULTS,
+                                                              DeleteRulesAction.REMOVE_ADD_DEFAULTS]) {
+            amountOfDeletedRules = data.rulesDeleted + multiTableFlowRulesCount
+        } else {
+            amountOfDeletedRules = data.rulesDeleted
+        }
```

Comment:
This is a monster, 80% of code is different for single/multi versions, test must be transparent. Split into two tests:  single-switch mode and multi-switch mode versions.

Reply:
done in d126fef361a0f71a5a50c48c47f88f03d56b073f